[WIP] docs(aio):edit architecture doc

jbogarthyde opened the pull request:
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: N/A


## What is the new behavior?


## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information
@StephenFluin @robwormald  Please check out proposed reorg/rewrite of architecture page - we should meet and talk about it.

jbogarthyde committed:
docs(aio): re-org and edit architecture doc

jbogarthyde committed:
docs(aio): re-org and edit architecture doc

mary-poppins commented:
You can preview 68c94fb at https://pr20248-68c94fb.ngbuilds.io/.

mary-poppins commented:
You can preview 6a3109a at https://pr20248-6a3109a.ngbuilds.io/.

gkalpak commented on aio/content/navigation.json:
```diff
@@ -127,7 +127,24 @@
         {
           "url": "guide/architecture",
```
We have decided that it is bad UX to have URLs on expanding entries. This should be moved to child entry (e.g. "overview" or something).

mary-poppins commented:
You can preview 8702625 at https://pr20248-8702625.ngbuilds.io/.

mary-poppins commented:
You can preview 2fe2acf at https://pr20248-2fe2acf.ngbuilds.io/.

mary-poppins commented:
You can preview 0cd478f at https://pr20248-0cd478f.ngbuilds.io/.

mary-poppins commented:
You can preview f45fe38 at https://pr20248-f45fe38.ngbuilds.io/.

jbogarthyde commented:
Note that latest commits did not make it into the preview

gkalpak commented:
> Note that latest commits did not make it into the preview

What do you mean, @jbogarthyde?
Which commits didn't make it?

jbogarthyde commented:
Looks like it’s all there now. Took a long time.

chuckjaz committed:
docs(compiler): document the $any type cast function (#20968)

Closes #20966

PR Close #20968

mary-poppins commented:
You can preview 68c94fb at https://pr20248-68c94fb.ngbuilds.io/.
You can preview 6a3109a at https://pr20248-6a3109a.ngbuilds.io/.
You can preview 8702625 at https://pr20248-8702625.ngbuilds.io/.
You can preview 2fe2acf at https://pr20248-2fe2acf.ngbuilds.io/.
You can preview 0cd478f at https://pr20248-0cd478f.ngbuilds.io/.
You can preview f45fe38 at https://pr20248-f45fe38.ngbuilds.io/.
You can preview 63513de at https://pr20248-63513de.ngbuilds.io/.
You can preview a4b3ab1 at https://pr20248-a4b3ab1.ngbuilds.io/.

mary-poppins commented:
You can preview d12d50d at https://pr20248-d12d50d.ngbuilds.io/.

Commit:
fix(language-service): Clear caches when program changes (#21337)

This commit fixes a bug whereby the caches are not cleared when the
program changes. This subsequently produces the incorrect error of
'Component ... is not included in a module ...'.

PR Close #19405

PR Close #21337

ocombe committed:
test(common): make date pipe tests work in more timezones (#21379)

Fixes #21112
PR Close #21379

gkalpak committed:
fix(aio): correctly handle redirects (#21416)

- Fixes handling of some redirects by the ServiceWorker.
- Fixes redirect for old `NgFor` to new `NgForOf` URL.

Fixes #21318

PR Close #21416

petebacondarwin committed:
style(aio): enforce strict TypeScript checks (#21342)

Closes #20646

PR Close #21342

gkalpak committed:
ci(aio): run e2e tests in `production` mode (#21470)

This will enable catching errors introduced by build optimizations that
do not appear in `development` mode.

Fixes #21446

PR Close #21470

alexeagle committed:
ci: disable integration/cli-hello-world test (#21492)

it is non-hermetic and breaks often due to unpinned dependencies.

PR Close #21492

ocombe committed:
fix(common): set correct timezone for ISO8601 dates in Safari (#21506)

Fixes #21491
PR Close #21506

jbogarthyde committed:
Merge branch 'jb-architecture-edit' of https://github.com/ExpertSupport/angular into jb-architecture-edit

StephenFluin reviewed (commented):
LGTM
